src/jobTree/jobStores/awsJobStore.py

```diff
@@ -241,8 +241,12 @@ def writeFileStream( self, jobStoreID ):
     def writeSharedFileStream( self, sharedFileName ):
         assert self._validateSharedFileName( sharedFileName )
         jobStoreFileID = self._newFileID( sharedFileName )
-        oldVersion = self._getFileVersion( jobStoreFileID )
-        with self._uploadStream( jobStoreFileID ) as (writable, key):
+        oldVersion, bucket = self._getFileVersion( jobStoreFileID )
```

Comment:
Sorry for the late change of heart. I know I asked you to do the assert for each call site of `_getFileVersion` but I'm seeing a potential refactoring that could reduce the verbosity. Rename  `_getFileVersion` to `_getFileVersionAndBucket`. Then add a `_getFileVersion( jobStoreFileId, bucket )` method that calls `_getFileVersionAndBucket`, does the conditional assert from here and then just returns the version.